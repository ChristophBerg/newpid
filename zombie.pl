#!/usr/bin/perl

# This script forks twice, and the middle process exits immediately. The third
# process will then gets pid 1 as parent, and exit later. At that point, pid 1
# receives a SIGCHLD signal. The first process keeps running so we can actually
# observe this effect.

if (fork == 0) {
	if (fork == 0) {
		sleep 1;
		exit 1;
	} else {
		exit 2;
	}
} else {
	sleep 2;
	system "ps xf";
	exit 3;
}
