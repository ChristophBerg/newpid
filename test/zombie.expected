  PID  PPID COMMAND
    1     0 newpid ./zombie.pl
    2     1 /usr/bin/perl ./zombie.pl
    3     2  \_ [zombie.pl] <defunct>
    5     2  \_ ps -opid,ppid,command xf
